core/services/directrequestocr/models.go

```diff
@@ -27,23 +29,36 @@ const (
 
 type RequestID [32]byte
 
+const RequestIDLength int = 32
+
+func (r *RequestID) Scan(value interface{}) error {
+	bytes, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("can't scan %T into RequestID", value)
+	}
+	if len(bytes) != RequestIDLength {
+		return fmt.Errorf("can't scan []byte of len %d into RequestID, want %d", len(bytes), RequestIDLength)
+	}
+	copy(r[:], bytes)
+	return nil
+}
+
+func (r RequestID) Value() (driver.Value, error) {
+	return r[:], nil
+}
+
 type Request struct {
-	ID            int64
-	RequestID     RequestID
-	RunID         int64
-	ReceivedAt    time.Time
-	RequestTxHash *common.Hash
-	State         RequestState
-	ResultReadyAt time.Time
-	Result        []byte
-	ErrorType     ErrType
-	Error         []byte
-	// True if this node submitted an observation for this request in any OCR rounds.
-	IsOCRParticipant  bool
+	RequestID         RequestID
+	RunID             *int64
+	ReceivedAt        time.Time
+	RequestTxHash     *common.Hash
+	State             RequestState
+	ResultReadyAt     *time.Time
```

Comment:
This is a pointer but ReceivedAt is not.

Reply:
Because ReceivedAt is NOT NULL in DB and therefore must be populated for a valid request.
ResultReadyAt may not be populated, hence pointer.